cmd/loom/loom.go

```diff
@@ -836,8 +842,18 @@ func loadApp(
 			}
 			return evm.NewLoomVm(state, eventHandler, receiptHandlerProvider.Writer(), createABM, cfg.EVMDebugEnabled), nil
 		})
+
+		ethDB := store.NewLoomEthDB(evmStore, nil)
+		evmRoot, _ := evmStore.Version()
+		stateDB := state.NewDatabase(ethDB)
+		stateDB.SetTrieDB(evmStore.TrieDB())
+		sdb, err := state.New(gcommon.BytesToHash(evmRoot), stateDB)
+		if err != nil {
+			return nil, err
+		}
+		evmState = loomchain.NewEVMState(evmStore, sdb)
```

Comment:
This stuff is pretty confusing, I think it'd be better to move this into `NewEVMState`:
```golang
evmState = loomchain.NewEVMState(evmStore)
```

Which will then look something like this:
```golang
func NewEVMState(evmStore *store.EvmStore) (*EVMState, error) {
  stateDatabase := state.NewDatabaseWithTrieDB(evmStore.TrieDB())
  evmRoot, _ := evmStore.Version()
  sdb, err := state.New(gcommon.BytesToHash(evmRoot), stateDatabase)
  if err != nil {
    return nil, err
  }
  return &EVMState{
    evmStore: evmStore,
    sdb:      sdb,
  }
}
```

Reply:
I can only do this.
```
func NewEVMState(evmStore *store.EvmStore) (*EVMState, error) {
	ethDB := store.NewLoomEthDB(evmStore, nil)
	stateDB := state.NewDatabase(ethDB).WithTrieDB(evmStore.TrieDB())
	evmRoot, _ := evmStore.Version()
	sdb, err := state.New(gcommon.BytesToHash(evmRoot), stateDB)
	if err != nil {
		return nil, err
	}
	return &EVMState{
		evmStore: evmStore,
		sdb:      sdb,
	}, nil
}
```